### DISCUSSION TITLE
Client side Internationalization middleware (static exports)

### DISCUSSION THREAD

**Discussion by FenderFred** [amannn/next-intl#323] [Q&A] [answered]
Hi,

Is it true that the middleware for localization can be running on NextJs SPA (output: export) ?
I see this in the documentation of Client side component:
https://next-intl-docs.vercel.app/docs/next-13/client-components#middlewarets

When I add the middleware without a locale in the path, the defaultLocale in the middleware do nothing.
Is it supposed to work ?

Thanks

**Comment by amannn** [comment, marked as answer]
Thanks for the question! I've created an issue from this discussion for visibility, as this is quite important: https://github.com/amannn/next-intl/issues/334

**Comment by Alenriquez96** [comment]
Any update on this?
